helper/resource/testing_config.go

```diff
@@ -101,10 +101,12 @@ func testStep(
 	}
 
 	// And another after a Refresh.
-	state, err = ctx.Refresh()
-	if err != nil {
-		return state, fmt.Errorf(
-			"Error on follow-up refresh: %s", err)
+	if !step.Destroy || (step.Destroy && !step.PreventPostDestroyRefresh) {
```

Comment:
Hmm... some of the code below (comment starting with "Data sources are tricky") is there with the intent that this wouldn't be necessary, but I assume when you were testing this you found that it still was for some reason?